File: Known.Web/src/Frame/Core/sys/SysFile.js
function SysFile() {
    //fields
    var url = {
        QueryModels: baseUrl + '/System/QueryFiles',
        DeleteModels: baseUrl + '/System/DeleteFiles',
        SaveModel: baseUrl + '/System/SaveFile',
        GetFileCategories: baseUrl + '/System/GetFileCategories',
        UploadFiles: baseUrl + '/System/UploadFiles',
        DownloadFile: baseUrl + '/System/DownloadFile',
        DownloadSysFile: baseUrl + '/System/DownloadSysFile'
    };

    var tree = new Tree('treeFile', {
        url: url.GetFileCategories, autoLoad: false,
        onClick: function (node) {
            view.loadGrid({ Category1: node.Category1, Category2: node.Category2 });
        }
    });

    var view = new View('SysFile', {
        url: url,
        left: tree,
        columns: [
            { field: 'Id', type: 'hidden' },
            { title: Language.Category1, field: 'Category1', width: '100px', type: 'text', required: true, onlyForm: true },
            { title: Language.Category2, field: 'Category2', width: '100px', type: 'text', onlyForm: true },
            {
                title: Language.FileName, field: 'Name', width: '100px', query: true, format: function (d) {
                    var icon = Utils.getExtIcon(d.ExtName);
                    var html = $('<span>').append('<i class="' + icon + '">');
                    $('<span>')
                        .addClass('link')
                        .html(d.Name)
                        .data('id', d)
                        .appendTo(html)
                        .on('click', function () {
                            var data = $(this).data('id');
                            if (data.Id === '') {
                                Ajax.download(url.DownloadSysFile, { bizId: data.BizId });
                            } else {
                                Ajax.download(url.DownloadFile, { id: data.Id });
                            }
                        });
                    return html;
                }, type: 'text', required: true, inputBlock: true
            },
            { title: Language.FileType, field: 'Type', width: '100px', type: 'text' },
            { title: Language.FileSize, field: 'Size', width: '100px', align: 'right', type: 'text', readonly: true },
            { title: Language.FileSourceName, field: 'SourceName', width: '100px', type: 'text', readonly: true, inputBlock: true },
            { title: Language.ExtName, field: 'ExtName', width: '100px', query: true, type: 'text', readonly: true },
            { title: Language.Note, field: 'Note', type: 'textarea', lineBlock: true },
            { title: Language.UploadBy, field: 'CreateBy', width: '100px' },
            { title: Language.UploadTime, field: 'CreateTime', width: '140px', placeholder: DateTimeFormat, align: 'center' }
        ],
        gridOption: {
            toolbar: {
                add: function (e) {
                    _addFile();
                },
                edit: function (e) {
                    e.editRow(function (d) {
                        if (d.row.Id === '') {
                            Layer.tips(Language.SysFileNotEdit);
                            return false;
                        }

                        return true;
                    });
                },
                remove: function (e) {
                    var exist = false;
                    e.selectRows(function (e) {
                        for (var i = 0; i < e.rows.length; i++) {
                            if (e.rows[i].Id === '') {
                                exist = true;
                                break;
                            }
                        }
                    });

                    if (exist) {
                        Layer.tips(Language.SysFileNotDelete);
                        return false;
                    }

                    e.deleteRows(url.DeleteModels, function (e) {
                        e.grid.reload();
                    });
                }
            }
        }
    });

    //methods
    this.render = function (dom) {
        view.render().appendTo(dom);
    }

    this.mounted = function () {
        view.load();
    }

    //private
    function _addFile() {
        var formAdd = new Form('AddFile', {
            style: 'form form-block',
            fields: [
                { title: Language.Category1, field: 'Category1', type: 'text', required: true },
                { title: Language.Category2, field: 'Category2', type: 'text' },
                { title: Language.Attachment, field: 'Files', type: 'file', multiple: true }
            ]
        });
        Layer.open({
            title: Language.AddDocument,
            width: 500, height: 250,
            component: formAdd,
            buttons: [{
                text: Language.OK, handler: function (e) {
                    if (!formAdd.validate())
                        return;

                    var data = formAdd.getData();
                    Ajax.upload('Files', url.UploadFiles, data, function () {
                        e.close();
                        view.load();
                    });
                }
            }]
        });
    }
}

$.extend(Page, {
    SysFile: { component: new SysFile() }
});

File: Shared/SysFile.cs
namespace Known;

/// <summary>
/// 系统文件实体类。
/// </summary>
[DisplayName("系统文件")]
public class SysFile : EntityBase
{
    /// <summary>
    /// 取得或设置一级分类。
    /// </summary>
    [Required]
    [MaxLength(50)]
    [DisplayName("一级分类")]
    public string Category1 { get; set; }

    /// <summary>
    /// 取得或设置二级分类。
    /// </summary>
    [MaxLength(50)]
    [DisplayName("二级分类")]
    public string Category2 { get; set; }

    /// <summary>
    /// 取得或设置文件名称。
    /// </summary>
    [Required]
    [MaxLength(250)]
    [DisplayName("文件名称")]
    public string Name { get; set; }

    /// <summary>
    /// 取得或设置文件类型。
    /// </summary>
    [MaxLength(50)]
    [DisplayName("文件类型")]
    public string Type { get; set; }

    /// <summary>
    /// 取得或设置文件路径。
    /// </summary>
    [MaxLength(500)]
    [DisplayName("文件路径")]
    public string Path { get; set; }

    /// <summary>
    /// 取得或设置文件大小。
    /// </summary>
    [Required]
    [DisplayName("文件大小")]
    public long Size { get; set; }

    /// <summary>
    /// 取得或设置原文件名。
    /// </summary>
    [Required]
    [MaxLength(250)]
    [DisplayName("源文件名")]
    public string SourceName { get; set; }

    /// <summary>
    /// 取得或设置扩展名。
    /// </summary>
    [Required]
    [MaxLength(50)]
    [DisplayName("扩展名")]
    public string ExtName { get; set; }

    /// <summary>
    /// 取得或设置备注。
    /// </summary>
    [DisplayName("备注")]
    public string Note { get; set; }

    /// <summary>
    /// 取得或设置业务ID。
    /// </summary>
    [MaxLength(250)]
    [DisplayName("业务ID")]
    public string BizId { get; set; }

    /// <summary>
    /// 取得或设置文件缩略图路径。
    /// </summary>
    [MaxLength(500)]
    [DisplayName("缩略图路径")]
    public string ThumbPath { get; set; }
}